lib/spack/spack/build_systems/perl.py

```diff
@@ -2,73 +2,87 @@
 # Spack Project Developers. See the top-level COPYRIGHT file for details.
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
-
-
 import inspect
 import os
 
 from llnl.util.filesystem import filter_file
 
-from spack.directives import extends
-from spack.package_base import PackageBase, run_after
+import spack.builder
+import spack.package_base
+from spack.directives import build_system, extends
+from spack.package_base import PackageBase
 from spack.util.executable import Executable
 
+from ._checks import BaseBuilder, execute_build_time_tests
+
 
 class PerlPackage(PackageBase):
-    """Specialized class for packages that are built using Perl.
+    """Specialized class for packages that are built using Perl."""
+
+    #: This attribute is used in UI queries that need to know the build
+    #: system base class
+    build_system_class = "PerlPackage"
+    #: Legacy buildsystem attribute used to deserialize and install old specs
+    legacy_buildsystem = "perl"
+
+    build_system("perl")
+
+    extends("perl", when="build_system=perl")
 
-    This class provides four phases that can be overridden if required:
 
-        1. :py:meth:`~.PerlPackage.configure`
-        2. :py:meth:`~.PerlPackage.build`
-        3. :py:meth:`~.PerlPackage.check`
-        4. :py:meth:`~.PerlPackage.install`
+@spack.builder.builder("perl")
+class PerlBuilder(BaseBuilder):
+    """The perl builder provides four phases that can be overridden, if required:
+
+        1. :py:meth:`~.PerlBuilder.configure`
+        2. :py:meth:`~.PerlBuilder.build`
+        3. :py:meth:`~.PerlBuilder.check`
+        4. :py:meth:`~.PerlBuilder.install`
 
     The default methods use, in order of preference:
         (1) Makefile.PL,
         (2) Build.PL.
```

Comment:
Would it make sense to split this into 2 separate build systems? (in a follow-up PR, not this PR)